roles/bmc_fw_update/tasks/main.yml

```
@@ -59,7 +59,9 @@
   delegate_to: localhost
 
 - name: Validate task was completed
-  ansible.builtin.assert: { that: "update_progress.redfish_facts.update_status.status == 'Completed'" }
+  ansible.builtin.fail:
+    msg: "Task failed with status: {{ update_progress.redfish_facts.update_status.status }}"
```

We can add this message in task/vars.yml
`bmc_fw_update_task_failure: "Task failed with status: %s"` and refer here as below, which makes code cleaner and maintainable.
```suggestion
    msg: "{{ bmc_fw_update_task_failure | format(update_progress.redfish_facts.update_status.status) }}"
```

---

thanks, done